source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/connect/ServerTypeSelectControl.java

```diff
@@ -114,33 +157,112 @@ public void widgetSelected(final SelectionEvent e) {
+    }
 
-        if (serverURI != null) {
-            serverControl.setServerURI(serverURI);
-        } else if (serverControl.getServerURI() != null) {
-            setServer(serverControl.getServerURI());
-        }
-        serverControl.setEnabled(!serverReadonly);
-        serverControl.addServerSelectionChangedListener(new ISelectionChangedListener() {
+    private Composite setupServerCombo(final GridLayout parentLayout) {
+        final Composite container = new Composite(this, SWT.NULL);
+        final GridLayout layout = SWTUtil.gridLayout(container, 2, false, 0, 0);
+        layout.marginWidth = 0;
+        layout.marginHeight = 0;
+        layout.horizontalSpacing = getHorizontalSpacing();
+        layout.verticalSpacing = getVerticalSpacing();
+
+        serverCombo = new Combo(container, SWT.NONE);
+        // setup Auto Complete for the combo box
+        proposalProvider = new PropsalProvider();
+        final ComboContentAdapter comboContentAdapter = new ComboContentAdapter();
+        serverComboAdapter = new ContentProposalAdapter(serverCombo, comboContentAdapter, proposalProvider, null, null);
+        serverComboAdapter.setProposalAcceptanceStyle(ContentProposalAdapter.PROPOSAL_REPLACE);
+        GridDataBuilder.newInstance().hGrab().hFill().applyTo(serverCombo);
+        serverCombo.addModifyListener(new ModifyListener() {
             @Override
-            public void selectionChanged(final SelectionChangedEvent event) {
+            public void modifyText(final ModifyEvent e) {
                 if (!ignoreServerChangeEvents) {
-                    setServer(serverControl.getServerURI());
+                    setServer(getServerFromCombo(), false);
                 }
             }
         });
 
-        // fire the buttons changed event so that the initial state is updated
-        onButtonsChanged();
+        final Button serversButton =
+            SWTUtil.createButton(container, Messages.getString("ServerSelectControl.ServersButtonText")); //$NON-NLS-1$
+        serversButton.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(final SelectionEvent e) {
+                onServersButtonSelected();
+            }
+        });
+
+        GridDataBuilder.newInstance().hSpan(parentLayout).hIndent(getHorizontalSpacing() * 4).hFill().hGrab().vIndent(
+            getVerticalSpacing()).applyTo(container);
+
+        return container;
+    }
+
+    private void populateServersCombo() {
+        final ServerList serverList =
+            ServerListManagerFactory.getServerListProvider(DefaultPersistenceStoreProvider.INSTANCE).getServerList();
+
+        int i = 0;
+        final Set<ServerListConfigurationEntry> servers = serverList.getServers();
+        final String[] list = new String[servers.size()];
+        for (final ServerListConfigurationEntry entry : servers) {
+            list[i++] = entry.getURI().toString();
+        }
+        Arrays.sort(list);

```

Comment:
Shouldn't we use Collections.sort(list,Collator.getInstance()) here to use locale depending sort order? *list* should be List<String> then.